code/game/objects/effects/chem/puddle.dm

```diff
@@ -0,0 +1,107 @@
+
+/obj/effect/decal/cleanable/chempuddle
+	name = "puddle"
+	icon = 'icons/effects/chempuddle.dmi'
+	icon_state = "puddle"
+
+	mouse_opacity = FALSE
+
+/obj/effect/decal/cleanable/chempuddle/Initialize()
+	. = ..()
+
+	create_reagents(100)
+
+	addtimer(CALLBACK(src, .proc/evaporation_pulse), 10 SECONDS)
+
+/obj/effect/decal/cleanable/chempuddle/Destroy()
+	var/turf/T = get_turf(src)
+
+	. = ..()
```

Comment:
Should [the rest of the Destroy logic] happen before we finish cleaning up object-specific stuff? (The for loop below this)

Reply:
This puddle needs to be [Destroyed] before the loop, as it is to ensure they don't look like they're connected to a puddle section that's not there.